src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/Segment.scala

```diff
@@ -42,15 +44,15 @@ object Segment {
    *
    * @return an Segment instance, fully repaired and usable
    */
-  def apply(location: File): Segment = {
+  def apply(siriusConfig: SiriusConfiguration, location: File): Segment = {
     location.mkdirs()
 
     val dataFile = new File(location, "data")
     val indexFile = new File(location, "index")
     val compactionFlagFile = new File(location, "keys-collected")
     val internalCompactionFlagFile = new File(location, "internally-compacted")
 
-    val data = UberDataFile(dataFile.getAbsolutePath)
+    val data = UberDataFile(dataFile.getAbsolutePath, siriusConfig)
```

Comment:
Maybe let `UberFileDesc` take carer off the config & produce the actual file object?
Feels a little strange having the config just for another layer to pass through.

Reply:
That's not a bad idea.